pkg/operator/ceph/object/bucket/predicate.go

```diff
@@ -62,8 +64,21 @@ func predicateController(ctx context.Context, c client.Client) predicate.Funcs {
 						}
 					}
 				}
-			}
 
+			case *cephv1.CephObjectStore:
+				objNew := e.ObjectNew.(*cephv1.CephObjectStore)
+				if objOld.Spec.IsExternal() {
+					if objOld.Spec.Gateway.ExternalRgwEndpoints[0] != objNew.Spec.Gateway.ExternalRgwEndpoints[0] {
+						logger.Debug("there is an update event on CephObjectStore CR, so reconcile the bucket provisioner")
+						// We reload the manager so that the controller restarts and goes
+						// through the CreateFunc again. Then the CephCluster watcher will be triggered
+						controller.ReloadManager()
```

Comment:
This will be quite disruptive, interrupting all reconciles and effectively restarting the operator. This approach is still concerning. I thought we didn't need this approach based on Blaine's design suggestion.

Reply:
I guess removing this should still work`controller.ReloadManager()`, we just need to re-run the object provisioner controller

Reply:
I will test